### DISCUSSION TITLE
add additional details to commit when merging a group of deps

### DISCUSSION THREAD

**Discussion by wyardley** [renovatebot/renovate#24818] [Suggest an Idea]
**What would you like Renovate to be able to do?**
When Renovate merges a group of dependencies (e.g., when you're grouping patch releases), currently, the commit message is just
```
 fix(deps): update patchreleases
```

I'd like to propose that an ASCII summary, similar to the one in the PR description, be added as well, so that it would become
```
 fix(deps): update patchreleases

- update foobar from ==1.16.28 -> ==1.16.47
- update whatsit from ==2.6.0 -> ==2.6.1
```

**Did you already have any implementation ideas?**

@wyardley

**Comment by rarkins** [comment]
Can you check out if [commitBodyTable](https://docs.renovatebot.com/configuration-options/#commitbodytable) satisfies your requirements?

**Comment by wyardley** [comment]
> Can you check out if commitBodyTable satisfies your requirements?

Sorry that I somehow missed that! I may have come across it before at some point; not sure why we don't have it enabled. I will try that out. Sorry for the noise, and thanks (as always) for the prompt response.

Personally, I would like if there were a way to have it as a bulleted list, but having the information in table format should be fine as well.

**Comment by rarkins** [comment]
We should be able to make commitBody template-able enough for that purpose too. I haven't experimented with it in a long while

**Comment by rarkins** [comment]
Next steps: verify if `commitBody` accepts templates and if the desired information is passed to it.

**Comment by wyardley** [comment]
btw, curious if there's any new information on how these templates work or if users can define their own.

I would love to see a different format (bulleted), as well as deduplication in the case where the same dependency is updated multiple times. For example:

instead of:
```
    | datasource         | package               | from   | to     |
    | ------------------ | --------------------- | ------ | ------ |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google-beta | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google-beta | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google      | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google-beta | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google-beta | 4.25.0 | 4.26.0 |
    | terraform-provider | hashicorp/google-beta | 4.25.0 | 4.26.0 |
```

Maybe something like this?
```
terraform-provider:
- Update hashicorp/google from 4.25.0 to 4.26.0
- Update hashicorp/google-beta from 4.25.0 to 4.26.0
 ```

**Reply by felipecrs** [reply]
There is no reason to keep duplicates in the table indeed. If Renovate can deduplicate the table rows by default, that'd be great already.

**Comment by adam-moss** [comment]
Bit of an old one but are there any plans to support this customisation?  We have a lot of teams using conventional commits and commitlint, and it isn't unusual to break the body-max-length default of 100 chars.

**Reply by rarkins** [reply]
What's an example commit body you'd want to see? I haven't thought about this in a long time but we may not be far away using templating

**Reply by adam-moss** [reply]
Haven't really considered how it should look tbh, it is more keeping within the 100 char limit.

Using a failing example:

```plaintext
| datasource | package                                             | from         | to                  |
| ---------- | --------------------------------------------------- | ------------ | ------------------- |
| docker     | eclipse-temurin                                     | 17-jdk-jammy | 17.0.10_7-jdk-jammy |
| maven      | au.com.dius.pact.consumer:junit5                    | 4.6.3        | 4.6.7               |
| maven      | org.springframework.boot:spring-boot-starter-parent | 3.2.1        | 3.2.3               |
```

being able to drop columns, e.g. datasource, or remove preceding "stuff" from package names would suffice.

That said, the other suggestion

```plaintext
terraform-provider:
- Update hashicorp/google from 4.25.0 to 4.26.0
- Update hashicorp/google-beta from 4.25.0 to 4.26.0
```

would work too :)

**Reply by wyardley** [reply]
Being able to dedupe identical items (as mentioned above) and to drop columns would be the most helpful from my standpoint. There are some cases where it's useful to know "I updated x 15 times", but in many other cases, it's less interesting / useful (I guess being able to summarize the count could also be item in a perfect world).

I could see some cases (like with Docker images or GitHub sources) where being able to do regex or substring replacement on the package name or version string would also help in terms of minimizing length.

**Reply by rarkins** [reply]
Regex replace for substrings is already supported in our handlebars templates

**Reply by adam-moss** [reply]
So does say editing the packageName carry through to includeBodyTable?  It isn't obvious if it does (although I must admit I haven't stress tested it).

**Reply by rarkins** [reply]
No, you'd need to make your own commit body using handlebars
